src/zookeeper_ext.rs

```diff
@@ -2,12 +2,22 @@ use acl::*;
 use consts::{CreateMode, ZkError};
 use zookeeper::{ZkResult, ZooKeeper};
 use std::iter::once;
+use std::collections::VecDeque;
 
 /// Extended ZooKeeper operations that are not needed for the "core."
 pub trait ZooKeeperExt {
     /// Ensure that `path` exists and create all potential paths leading up to it if it does not.
     /// This operates in a manner similar to `mkdir -p`.
     fn ensure_path(&self, path: &str) -> ZkResult<()>;
+
+    /// Performs a breadth-first tree traversal of the tree starting at `path`,
+    /// returning a list of fully prefixed child nodes.
+    /// *NOTE*: This is not an atomic operation.
+    fn get_children_recursive(&self, path: &str) -> ZkResult<Vec<String>>;
+
+    /// Deletes the node at `path` and all its children.
+    /// *NOTE*: This is not an atomic operation.
+    fn delete_recursive(&self, path: &str) -> ZkResult<()>;
```

Comment:
I know the Java client's `ZKUtil.deleteRecursive` isn't atomic, but it was written before multi-key transactions existed. It is now possible to make this recursive by calling `get_children_recursive`, then building a big multi of deletes from the reversed list.

Reply:
@ekmartin do you think you can change it to support the multi key tx way?

Reply:
That sounds like a good idea. Wouldn't that mean rust-zookeeper would need transaction support first though (#4)? If I'm understanding this correctly we'd need something similar to the Java client's [`multi`](https://github.com/apache/zookeeper/blob/master/src/java/main/org/apache/zookeeper/ZooKeeper.java#L1732-L1737) method, and possibly something like the [`Transaction`](https://github.com/apache/zookeeper/blob/master/src/java/main/org/apache/zookeeper/Transaction.java) builder - which seems like a wrapper on top of `multi`. That would also involve creating something similar to [`Op`](https://github.com/apache/zookeeper/blob/master/src/java/main/org/apache/zookeeper/Op.java), and finally, support for serializing and deserializing the multi requests and responses - similar to [`MultiTransactionRecord`](https://github.com/apache/zookeeper/blob/df5519ab9dac9940f35cc4b308b560f2603aec7f/src/java/main/org/apache/zookeeper/MultiTransactionRecord.java) and [`MultiResponse`](https://github.com/apache/zookeeper/blob/master/src/java/main/org/apache/zookeeper/MultiResponse.java).

I wouldn't mind helping implement that, but might be better to separate it into a separate PR and possibly discuss the design first?